### PR TITLE
replace code with pdb2sql package

### PR DESCRIPTION
**Pull request by CunliangGeng**
To solve issues: #100 #102.

**Note: please first merge RP #103 and #107 and then this RP.**

### CODE REVIEW

**Comment by CunliangGeng** [comment]
@NicoRenaud All checks failed because the current `pdb2sql` package in pip is still the old one. BTW, all tests passed at local machines using the latest `pdb2sql` package.

File: deeprank/features/AtomicFeature.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def __init__(self, pdbfile, param_charge=None, param_vdw=None,
         self.atom_key = 'chainID, resSeq, resName, name'
 
         # read the pdb as an sql
-        self.sqldb = pdb2sql(self.pdbfile)
+        self.sqldb = pdb2sql.pdb2sql(self.pdbfile)
```
Comment (NicoRenaud) [inline deeprank/features/AtomicFeature.py]: that's from pdb2sql_core right ?

File: deeprank/features/FeatureClass.py [hunk 2]
```diff
@@ -71,7 +71,11 @@ def export_data_hdf5(self, featgrp):
                 # append
                 ds.append(feat)
 
-            ds = np.array(ds).astype('|S' + str(len(ds[0])))
+            if ds:
+                ds = np.array(ds).astype('|S' + str(len(ds[0])))
+            else:
+                ds = np.array(ds)
```
Comment (NicoRenaud) [inline deeprank/features/FeatureClass.py]: good catch !

File: deeprank/generate/DataGenerator.py [hunk 3]
```diff
@@ -411,11 +413,11 @@ def create_database(
                     self.f5.copy(mol_name + '/features/', molgrp)
 
                     # rotate the feature
-                    self._rotate_feature(molgrp, axis, angle, mol_center)
+                    DataGenerator._rotate_feature(molgrp, axis, angle, mol_center)
```
Comment (NicoRenaud) [inline deeprank/generate/DataGenerator.py]: why not self here ?

File: deeprank/generate/DataGenerator.py [hunk 4]
```diff
@@ -577,7 +579,7 @@ def add_feature(self, remove_error=True, prog_bar=True):
                     aug_molgrp.create_dataset("features/" + k, data=data)
 
                     # rotate
-                    self._rotate_feature(
+                    DataGenerator._rotate_feature(
```
Comment (NicoRenaud) [inline deeprank/generate/DataGenerator.py]: same here, why DataGenerator. and not self._rotate_feature ?

File: deeprank/generate/DataGenerator.py [hunk 5]
```diff
@@ -1330,48 +1337,23 @@ def _add_aug_pdb(molgrp, pdbfile, name, axis, angle):
...
-            except BaseException:
-                line += '{: 6.2f}'.format(0)    # occ
-                line += '{: 6.2f}'.format(0)    # temp
-            data.append(line)
+        # get molecule center
+        xyz = sqldb.get('x,y,z')
+        center = np.mean(xyz, 0)
 
+        # get the pdb-format data
+        data = sqldb.sql2pdb()
```
Comment (NicoRenaud) [inline deeprank/generate/DataGenerator.py]: nice little function here :)

File: deeprank/features/AtomicFeature.py (hunk 1 above)
Comment (CunliangGeng) [reply deeprank/features/AtomicFeature.py]: yes

File: deeprank/generate/DataGenerator.py (hunk 5 above)
Comment (CunliangGeng) [reply deeprank/generate/DataGenerator.py]: Thanks :-)

File: deeprank/generate/DataGenerator.py (hunk 3 above)
Comment (CunliangGeng) [reply deeprank/generate/DataGenerator.py]: I misunderstood static method before. self is better here.

File: deeprank/generate/DataGenerator.py (hunk 4 above)
Comment (CunliangGeng) [reply deeprank/generate/DataGenerator.py]: Updated now.

**Pull request merged**